CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -361,12 +361,15 @@ public static function validateFinancialAccount($financialAccountId, $financialA
    * with Financial Account
    *
    * @param array $params
+   *   Holds submitted formvalues and params from api for updating/adding contribution
    *
    * @param int $contributionID
+   *   Contribution ID
    *
    * @param obj $form
+   *   Holds form object of CRM_Contribute_Form_Contribution
    *
-   * @return string
+   * @return bool
    *
    */
   public static function checkFinancialTypeHasDeferred($params, $contributionID = NULL, $form = NULL) {
```

Comment:
I'm confused what you mean by 'the former': this is 'acceptable practice' to read but not modify contents of the form? Or did you mean 'the latter', the form should do the work with the Priceset so that more detailed arg than the form would be passed into this function. Thanks for the feedback @eileenmcnaughton